include/deal.II/base/polynomial_space.h

```diff
@@ -141,7 +141,9 @@ class PolynomialSpace
   void compute (const Point<dim>            &unit_point,
                 std::vector<double>         &values,
                 std::vector<Tensor<1,dim> > &grads,
-                std::vector<Tensor<2,dim> > &grad_grads) const;
+                std::vector<Tensor<2,dim> > &grad_grads = Utilities::EmptyVectorGenerator<Tensor<2,dim> >().get_reference(),
+                std::vector<Tensor<3,dim> > &grad_hessians = Utilities::EmptyVectorGenerator<Tensor<3,dim> >().get_reference(),
+                std::vector<Tensor<4,dim> > &grad_hessian_grads = Utilities::EmptyVectorGenerator<Tensor<4,dim> >().get_reference()) const;
```

Comment:
I must admit that I'm not too fond of the `EmptyVectorGenerator` solution. It relies on the fact that the lifetime of the default argument surrounds the lifetime of the call to this function, but assuming anything about the lifetime of temporaries always seems shady to me.

I suppose you do this because you want to stay backward compatible. What if you allowed yourself to break backward compatibility? Would you have to adjust a lot of different places?